packages/ember-data/lib/serializers/json_serializer.js

```diff
@@ -261,6 +261,16 @@ export default Ember.Object.extend({
   },
 
   /**
+    @method normalizeErrors
+    @private
+  */
+  normalizeErrors: function(type, hash) {
+    this.normalizeId(hash);
+    this.normalizeAttributes(type, hash);
+    this.normalizeRelationships(type, hash);
```

Comment:
Hey @ahacking,

We're updating our app to latest Ember-Data and we've been having issue in our errors processing because of this. When using polymorphic relationships, `normalizeRelationships` will inspect the hash for a `type` property. Since this is an error payload, the `type` property won't be set.

Here's a [workaround implementation](https://gist.github.com/jbourassa/4fe88271d271524402a9) which works great for us. Do you see any issue with it? I'm thinking of opening a PR with it.